File: gaia-sdk-javascript/src/graphql/request/type/SkillEvaluation.ts

import {SyncSkillEvaluation} from "./SyncSkillEvaluation";
import {AsyncSkillEvaluation} from "./AsyncSkillEvaluation";

import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {RuntimeState} from "../enumeration/RuntimeState";
import {SkillState} from "../enumeration/SkillState";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

export class SkillEvaluation extends Array<(_:VariableRegistry) => string> {
public _typeName = "SkillEvaluation";
    public syncEval = (impulse: string|undefined, config: (_:SyncSkillEvaluation) => void) => this.push((registry) => {
        const name1 = registry.register("impulse", impulse);
        const entity = new SyncSkillEvaluation();
        config(entity);
        return `syncEval(impulse:${name1}){` + entity.render(registry) + "}"
    });

    public asyncEval = (impulse: string|undefined, config: (_:AsyncSkillEvaluation) => void) => this.push((registry) => {
        const name1 = registry.register("impulse", impulse);
        const entity = new AsyncSkillEvaluation();
        config(entity);
        return `asyncEval(impulse:${name1}){` + entity.render(registry) + "}"
    });

    public render = (registry: VariableRegistry):String => this.map(e => e(registry)).join(" ");
}

File: gaia-sdk-python/gaia_sdk/graphql/request/type/SkillEvaluation.py

from gaia_sdk.graphql.request.type.SyncSkillEvaluation import SyncSkillEvaluation
from gaia_sdk.graphql.request.type.AsyncSkillEvaluation import AsyncSkillEvaluation

from typing import Callable, List
from gaia_sdk.api.VariableRegistry import VariableRegistry
from gaia_sdk.graphql.request.enumeration.Order import Order
from gaia_sdk.graphql.request.enumeration.OrderByField import OrderByField
from gaia_sdk.graphql.request.enumeration.EdgeOrderByField import EdgeOrderByField
from gaia_sdk.graphql.request.enumeration.EdgeType import EdgeType


class SkillEvaluation(list):

    def sync_eval(self, impulse: str, config: Callable[['SyncSkillEvaluation'], None]):
        def callback(registry: VariableRegistry):
            name1 = registry.register("impulse", impulse)
            entity = SyncSkillEvaluation()
            config(entity)
            return f'syncEval(impulse:{name1})' + '{' + entity.render(registry) + '}'
        self.append(callback)

    def async_eval(self, impulse: str, config: Callable[['AsyncSkillEvaluation'], None]):
        def callback(registry: VariableRegistry):
            name1 = registry.register("impulse", impulse)
            entity = AsyncSkillEvaluation()
            config(entity)
            return f'asyncEval(impulse:{name1})' + '{' + entity.render(registry) + '}'
        self.append(callback)

    def render(self, registry: VariableRegistry):
        return " ".join(map(lambda e: e(registry), self))
